### ISSUE TITLE
Feature request: Economy O2 target

### ISSUE THREAD

**Issue by RempageR1** [noisymime/speeduino#984]
Hi,

I`m using the Speeduino fuel only on my Vmax and want it to use the normal map during normal running (stop-go traffic, twisties etc) so it has quick throttle response, but use a higher AFR target when cruising the highway.

I think this might be accomplished by modifying the correctionAFRClosedLoop() function and adding something like this:
- If TPSdot is below some value for 30 seconds (steady throttle)
- If TPS is above 2% (we`re not idling)
- If O2 target is within some bounds (so the O2 sensor is working)
- Then set AFR target to 17 (or whatever value you set that the engine can run)
- Optionally, Allow the configPage6.egoLimit to be higher to ensure the lean AFR can be reached.

I will build this for my bike to see if I can improve the highway cruising economy a bit, however if you find it interesting, I would be happy to code it so it can be part of the Speeduino code. Just let me know if it`s something that might be useful. Thank you.

**Comment by RempageR1** [comment]
I understand you can normally tune those cells, but like I mentioned it impacts throttle responsiveness in my bike. If you are already on the very-lean site and them open the throttle quickly, the added fuel is not in time and the motor stumbles for a very short moment. Therefore I would like to develop this feature if it`s something that the community might want.

**Comment by digmorepaka** [comment]
I agree with @jackscr12, your cruising areas can be tuned for a leaner mixture from WOT, this seems to be more of a particular tune issue.

Actual economy tuning also isn't about leaning out as much as you can before knocking a hole in your piston, you have to do torque optimization on the dyno - that includes both fuel and ignition. 

> it impacts throttle responsiveness
I believe this is more of a problem with your AE settings. MAPVE tends to respond a bit behind TPSAE. If your TPSAE isn't kicking in early enough I'd also suggest taking a look at the TPS ADC filter option.

**Comment by adbancroft** [comment]
Closing - issue is too old to troubleshoot. Please reopen is this is still a problem.
